build-system/compile/debug-compilation-lifecycle.js

```diff
@@ -42,7 +42,7 @@ const LIFECYCLES = {
  *
  * @param {string} lifecycle
  * @param {string} fullpath
- * @param {string=} content
+ * @param {?string=} content
```

Comment:
This seems wrong because the JSDoc says the last two params have default values, but the function signature doesn't specify any.

I think it's best to make the last two params purely optional and drop the default specifier.

Reply:
The function signature does not specify a default, but it does assign a default value in the function body. Would this make more sense to you if I moved the default assignment to the signature?

Reply:
That wouldn't work because in the event `content` and `sourcemap` are not passed in, this function reads them from a file, and we don't want to have a file read operation in the function signature.

I think the suggestion I posted above is the cleanest way to do it (make the params strictly optional and drop the default). So, `@param {?string}` for both `content` and `sourcemap`. Would that take care of the type errors?

Reply:
I'm a little confused by this suggestion. `@param {?string}` is equivalent to `@param {string|null}` `@param {string=}` is intended to indicate the parameter is optional using closure syntax (https://jsdoc.app/tags-param.html#optional-parameters-and-default-values).

`@param {string=}` will result in a single type error in [esbuild-bable](https://github.com/ampproject/amphtml/blob/main/build-system/common/esbuild-babel.js#L60) where the variable `code` is of type `string | null | undefined`. While I could add an assertion there I think the usage is valid.

`@param {?string}` results in 6 type errors from the places where the function is called without 4 parameters.

Reply:
Aha, I see. Thanks for explaining this to me. I now stand corrected, carry on!